### ISSUE TITLE
Fix `property-no-deprecated` false positive for `-webkit-box-orient: vertical;`

### ISSUE THREAD

**Issue by silverwind** [stylelint/stylelint#8698]
https://github.com/stylelint/stylelint/pull/8682 has marked `-webkit-box-orient` as deprecated but there exists a standardized co-dependency of three specific webkit-prefixed properties which is not deprecated, specifically:

```css
div {
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 2;
}
```

Note about this [from MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/line-clamp):

> Despite these prefixed properties being deprecated, the co-dependency of these three properties is a fully specified behavior and will continue to be supported.

Therefor I suggest removing the `-webkit-box-orient` property from the deprecation list, or only flag it when it appears without the other two properties in the same selector.

**Comment by immitsu** [comment]
@silverwind  thanks for the report.

There is something to discuss here.

Formally, `-webkit-box-orient` is a deprecated property. `-webkit-line-clamp` is not deprecated, but for projects that are not aiming to support legacy browsers, it makes sense to use `line-clamp`. Nevertheless, as you noted, there is a co-dependency, and your suggestion is relevant.

Let's see what the team members suggest.

As a temporary solution, you can add `-webkit-box-orient` to `ignoreProperties`:
```json
{
  "property-no-deprecated": [true, { "ignoreProperties": ["-webkit-box-orient"] }]
}
```

I found [the following discussion](https://github.com/w3c/csswg-drafts/issues/10439), which seems like it might be useful in our case.

**Comment by silverwind** [comment]
FYI, all browsers implement `line-clamp` as `-webkit-line-clamp` only. There is no non-prefixed alternative to achieve this behavior of clamping a text block to a certain amount of lines and for it to work `-webkit-box-orient` has to be present. Note all the `-x-` here:

<img width="736" height="315" alt="Image" src="https://github.com/user-attachments/assets/af12be11-2c41-4ea7-85b6-3cd3382c4c3a" />

**Comment by Mouvedia** [comment]
Indeed `-webkit-box-orient` should not be marked as deprecated if its value is `vertical`.
That's an oversight on our part.

**Comment by silverwind** [comment]
Does `property-no-deprecated` support configuration for not flagging a property if it has a certain value? Does not seem like it at first glance.

**Comment by immitsu** [comment]
> Does `property-no-deprecated` support configuration for not flagging a property if it has a certain value? Does not seem like it at first glance.

At the moment - no, but we can add it as an optional secondary option. This is how it is done in [value-no-vendor-prefix](https://github.com/stylelint/stylelint/blob/main/lib/rules/value-no-vendor-prefix/README.md)

**Comment by Mouvedia** [comment]
> Does `property-no-deprecated` support configuration for not flagging a property if it has a certain value? Does not seem like it at first glance.

It won't be an option for now.
i.e. that would be a separate issue

**Comment by silverwind** [comment]
I'm fine with special-casing these cases in code for now. Configuration to specify key-value pairs might be nice for the future:

````json
"property-no-deprecated": [true, {"ignoreDeclarations": {"-webkit-box-orient": "vertical"}}]
````
